src/main/java/bio/terra/service/configuration/ConfigFaultCounted.java

```diff
@@ -0,0 +1,144 @@
+package bio.terra.service.configuration;
+
+import bio.terra.app.controller.exception.ValidationException;
+import bio.terra.model.ConfigFaultCountedModel;
+import bio.terra.model.ConfigFaultModel;
+import bio.terra.model.ConfigModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Random;
+
+public class ConfigFaultCounted extends ConfigFault {
+    private final Logger logger = LoggerFactory.getLogger(ConfigFaultCounted.class);
+
+    private final ConfigFaultCountedModel originalCountedModel;
+
+    // This state is synchronized because it is referenced in synchronized code and findbugs
+    // wants it to be consistently synchronized everywhere...
+    private ConfigFaultCountedModel countedModel;
+
+    // Access to this state must be synchronized so that multiple threads do not mess it up.
+    private boolean doneSkipping;
+    private int skippedCounter;
+    private boolean doneInserting;
+    private int insertCounter;
+    private Random random;
+    private int every;
+    private int everyCounter;
+    private int testCounter;
+
+    // -- Instance methods --
+    ConfigFaultCounted(ConfigEnum configEnum,
+                       ConfigFaultModel.FaultTypeEnum faultType,
+                       boolean enabled,
+                       ConfigFaultCountedModel countedModel) {
+        super(configEnum, faultType, enabled);
+        this.originalCountedModel = countedModel;
+        this.countedModel = countedModel;
+        initCounters();
+    }
+
+    @Override
+    synchronized ConfigModel get() {
+        ConfigFaultModel faultCounted = new ConfigFaultModel()
+            .faultType(ConfigFaultModel.FaultTypeEnum.SIMPLE)
+            .enabled(isEnabled())
+            .counted(countedModel);
+
+        return new ConfigModel()
+            .configType(ConfigModel.ConfigTypeEnum.FAULT)
+            .name(getName())
+            .fault(faultCounted);
+    }
+
+    @Override
+    synchronized void setFaultConfig(ConfigFaultModel faultModel) {
+        if (faultModel.getCounted() == null) {
+            throw new ValidationException("Set of a counted fault requires a counted fault model");
```

Comment:
Could you instead put this check in an overridden validate() that also calls the superclass' method?

Reply:
I made your suggested change.